core/java/iesi-core/src/test/java/io/metadew/iesi/common/crypto/FrameworkCryptoTest.java

```diff
@@ -0,0 +1,52 @@
+package io.metadew.iesi.common.crypto;
+
+import io.metadew.iesi.common.configuration.Configuration;
+import org.junit.Test;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+
+import java.io.ByteArrayInputStream;
+import java.util.Scanner;
+
+import static junit.framework.TestCase.assertEquals;
+
+@PrepareForTest(FrameworkCrypto.class)
+public class FrameworkCryptoTest {
+
+    @Test
+    public void testJavaKeystore() throws Exception {
+        String password = "foobar";
+        String currentDirectory = System.getProperty("user.dir");
+        String keystoreLocation = currentDirectory + "/src/test/resources/" + Configuration.getInstance().getMandatoryProperty("iesi.security.encryption.keystore-path").toString();
+
+        System.setIn(new ByteArrayInputStream(password.getBytes()));
+        Scanner scanner = new Scanner(System.in);
+        String userinput = scanner.nextLine();
+        String alias = "mypass";
+        String keyJKS = new JavaKeystore().loadKey(userinput.toCharArray(), keystoreLocation, alias);
+        assertEquals("c7c1e47391154a6a", keyJKS);
+    }
+
+    @Test
+    public void testJavaKeystoreWrongPassword() {
+        String password = "fooar";
+        String currentDirectory = System.getProperty("user.dir");
+        String keystoreLocation = currentDirectory + "/src/test/resources/" + Configuration.getInstance().getMandatoryProperty("iesi.security.encryption.keystore-path").toString();
+
+        System.setIn(new ByteArrayInputStream(password.getBytes()));
+        Scanner scanner = new Scanner(System.in);
+        String userinput = scanner.nextLine();
+        String alias = "mypass";
+        try {
+            new JavaKeystore().loadKey(userinput.toCharArray(), keystoreLocation, alias);
+        } catch (Exception e) {
+            assertEquals("java.io.IOException: Integrity check failed: java.security.UnrecoverableKeyException: Failed PKCS12 integrity checking", e.getMessage());
+        }
+    }
+
+    @Test
+    public void testKeyInConf() {
+        if (Configuration.getInstance().getProperty("iesi.security.encryption.key").isPresent()) {
+            assertEquals("c7c1e47391154a6a", Configuration.getInstance().getMandatoryProperty("iesi.security.encryption.key").toString());
+        }
```

Comment:
Does this actually test the conf?
Also, use something different than hardcoded password in Java Keystore to be sure key is not defaulted to hardcoded key